### PR TITLE
Document the versioning and release process

### PR DESCRIPTION
**Pull request by renmaosheng**
Document the versioning and release process.

Signed-off-by: Steven Ren <stevenr@stevenr-a01.vmware.com>

### CODE REVIEW

File: RELEASES.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+## Releases
+Harbor releases will be versioned using dotted triples, similar to [Semantic Version](http://semver.org/). For this specific document, we will refer to the respective components of this triple as `<major>.<minor>.<patch>`. The version number may have additional information, such as "-rc1,-rc2,-rc3" to mark release candidate builds for earlier access. Such releases will be considered as "pre-releases".
+
+### Major and Minor Releases
+Major and minor releases of Harbor will be branched from master when the release reaches to `RC(release candidate)` state. The branch format should follow `release/<major>.<minor>.0`. For example, once the release `v1.0.0` reaches to RC, a branch will be created with the format `release/1.0.0`. When the release reaches to `GA(General Available)` state, The tag with format `v<major>.<minor>.<patch>` and should be made with command `git tag -s v<major>.<minor>.<patch>`. The release cadency is around 3 months, might be adjusted based on open source event, but will communicate it clearly.
```
Comment (reasonerjt) [inline RELEASES.md]: The branch name looks like `release-1.0.0`

File: RELEASES.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+### Minor Release Support Matrix
+| Version | Supported          |
+| ------- | ------------------ |
+| Harbor v1.7.x   | :white_check_mark: |
+| Harbor v1.8.x   | :white_check_mark: |
+| Harbor v1.9.x   | :white_check_mark: |
+
+### Upgrade path and support policy
+The upgrade path for Harbor is (1) 1.0.x patch releases are always compatible with its major and minor version. (2) Harbor only supports two previous minor releases to upgrade to current minor release. For an example, 1.9.0 will only support 1.7.0 and 1.8.0 to upgrade from, 1.6.0 to 1.9.0 is not supported. One should upgrade to 1.8.0 first, then to 1.9.0.
```
Comment (reasonerjt) [inline RELEASES.md]: Don't quite understand item (1), and `For an example` better be rephrase to `For example`

**Comment by reasonerjt** [comment]
@renmaosheng 
Thanks a lot for the help!

File: RELEASES.md (hunk 2 above)
Comment (renmaosheng) [reply RELEASES.md]: (1) means users can always upgrade to newer patch version from the same minor release

File: RELEASES.md (hunk 1 above)
Comment (renmaosheng) [reply RELEASES.md]: thanks for the review. fixed it.

File: RELEASES.md (hunk 2 above)
Comment (renmaosheng) [reply RELEASES.md]: fixed for the typo, thanks.

**Pull request merged**